src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -744,32 +721,43 @@ def __convert_feature_dict_to_keyvalue_list(features_dict, enabled_for_keyword):
                 if isinstance(v, dict):
                     # This may be a conditional feature
                     feature_flag_value.enabled = False
-                    try:
-                        feature_flag_value.conditions = {'client_filters': v[enabled_for_keyword]}
-                    except KeyError:
-                        raise ValidationError("Feature '{0}' must contain '{1}' definition or have a true/false value. \n".format(str(k), enabled_for_keyword))
-
-                    if feature_flag_value.conditions["client_filters"]:
+                    feature_flag_value.conditions = {}
+                    enabled_for_found = False
+
+                    for condition, condition_value in v.items():
+                        if condition == feature_management_keywords.enabled_for:
+                            feature_flag_value.conditions[FeatureFlagConstants.CLIENT_FILTERS] = condition_value
+                            enabled_for_found = True
+                        elif condition == feature_management_keywords.requirement_type:
+                            if condition_value not in ("All", "Any"):
```

Comment:
Did we decide to keep the case sensitive comparison for Any/All?

Reply:
Actually I see in the provider that it is not a case-sensitive evaluation. https://github.com/microsoft/FeatureManagement-Dotnet/blob/aafe0c95a54a32b376d718580db9bc82c7567d33/src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs#L200C66-L200C140

I will update this accordingly.